src/dbus/dbus_message_helper_openthread.cpp

```diff
@@ -0,0 +1,366 @@
+
+    VerifyOrExit(dbus_message_iter_open_container(aIter, DBUS_TYPE_STRUCT, nullptr, &sub));
+    SuccessOrExit(err = ConvertToDBusMessage(&sub, args));
+    VerifyOrExit(dbus_message_iter_close_container(aIter, &sub) == true, err = OTBR_ERROR_DBUS);
+exit:
+    return err;
+}
+
+otbrError DBusMessageExtract(DBusMessageIter *aIter, otMacCounters &aValue)
+{
+    auto args = std::tie(
+        aValue.mTxTotal, aValue.mTxUnicast, aValue.mTxBroadcast, aValue.mTxAckRequested, aValue.mTxAcked,
+        aValue.mTxNoAckRequested, aValue.mTxData, aValue.mTxDataPoll, aValue.mTxBeacon, aValue.mTxBeaconRequest,
+        aValue.mTxOther, aValue.mTxRetry, aValue.mTxErrCca, aValue.mTxErrAbort, aValue.mTxErrBusyChannel,
+        aValue.mRxTotal, aValue.mRxUnicast, aValue.mRxBroadcast, aValue.mRxData, aValue.mRxDataPoll, aValue.mRxBeacon,
+        aValue.mRxBeaconRequest, aValue.mRxOther, aValue.mRxAddressFiltered, aValue.mRxDestAddrFiltered,
+        aValue.mRxDuplicated, aValue.mRxErrNoFrame, aValue.mRxErrUnknownNeighbor, aValue.mRxErrInvalidSrcAddr,
+        aValue.mRxErrSec, aValue.mRxErrFcs, aValue.mRxErrOther);
+    DBusMessageIter sub;
+    otbrError       err = OTBR_ERROR_NONE;
+
+    VerifyOrExit(dbus_message_iter_get_arg_type(aIter) == DBUS_TYPE_STRUCT, err = OTBR_ERROR_DBUS);
+    dbus_message_iter_recurse(aIter, &sub);
+    SuccessOrExit(err = ConvertToTuple(&sub, args));
+exit:
+    return err;
+}
+
+otbrError DBusMessageEncode(DBusMessageIter *aIter, const otIpCounters &aValue)
+{
+    DBusMessageIter sub;
+    otbrError       err  = OTBR_ERROR_NONE;
+    auto            args = std::tie(aValue.mTxSuccess, aValue.mRxSuccess, aValue.mTxFailure, aValue.mRxFailure);
+
+    VerifyOrExit(dbus_message_iter_open_container(aIter, DBUS_TYPE_STRUCT, nullptr, &sub));
+    SuccessOrExit(err = ConvertToDBusMessage(&sub, args));
+    VerifyOrExit(dbus_message_iter_close_container(aIter, &sub) == true, err = OTBR_ERROR_DBUS);
+exit:
+    return err;
+}
+
+otbrError DBusMessageExtract(DBusMessageIter *aIter, otIpCounters &aValue)
+{
+    DBusMessageIter sub;
+    otbrError       err  = OTBR_ERROR_NONE;
+    auto            args = std::tie(aValue.mTxSuccess, aValue.mRxSuccess, aValue.mTxFailure, aValue.mRxFailure);
+
+    VerifyOrExit(dbus_message_iter_get_arg_type(aIter) == DBUS_TYPE_STRUCT, err = OTBR_ERROR_DBUS);
+    dbus_message_iter_recurse(aIter, &sub);
+    SuccessOrExit(err = ConvertToTuple(&sub, args));
+exit:
+    return err;
+}
+
+otbrError DBusMessageEncode(DBusMessageIter *aIter, const otMasterKey &aValue)
+{
+    std::vector<uint8_t> data(&aValue.m8[0], &aValue.m8[0] + sizeof(aValue.m8));
+
+    return DBusMessageEncode(aIter, data);
+}
+
+otbrError DBusMessageExtract(DBusMessageIter *aIter, otMasterKey &aValue)
+{
+    otbrError            err;
+    std::vector<uint8_t> data;
+
+    SuccessOrExit(err = DBusMessageExtract(aIter, data));
+    VerifyOrExit(data.size() == sizeof(aValue.m8));
+    memcpy(aValue.m8, &data[0], sizeof(aValue.m8));
```

Comment:
`&data[0]` -> `masterKey.data`

Reply:
The same for function `otbrError DBusMessageEncode(DBusMessageIter *aIter, const otPskc &aValue)`